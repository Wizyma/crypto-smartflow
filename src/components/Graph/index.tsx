import React, { useMemo } from 'react';
import { PieChart, Pie, Tooltip, Cell } from 'recharts';
import { Card, Text, Divider } from '@zeit-ui/react';

import useTotalAmount from 'hooks/useTotalAmount';
import useGraphColor from 'hooks/useGraphColor';
import { CryptoRsponse } from 'types/dataContract';
import { CRYPTO_DICTIONNARY } from '../../constatns';

export default function Graph({
  cryptos,
  wallet,
}: {
  cryptos: CryptoRsponse;
  wallet: { [key: string]: number };
}) {
  const { total, percentage } = useTotalAmount({ cryptos, wallet });
  const pieData = useMemo(
    () =>
      percentage.map((item) => {
        const key = Object.keys(item)[0];

        return {
          name: CRYPTO_DICTIONNARY[key],
          value: parseFloat(item[key]),
        };
      }),
    [percentage],
  );
  const handleColor = useGraphColor();

  return (
    <>
      <Card.Content>
        <PieChart width={800} height={800}>
          <Pie
            isAnimationActive
            data={pieData}
            dataKey="value"
            fill="#8884d8"
            label={({ value }) => `${value} %`}
          >
            {pieData.map((entry, index) => (
              <Cell key={`cell-${index}`} fill={handleColor(entry.value)} />
            ))}
          </Pie>
          <Tooltip formatter={(value) => `${value} %`} />
        </PieChart>
      </Card.Content>
      <Divider y={0} />
      <Card.Content>
        <Text style={{ color: '#ccc', textAlign: 'center', fontWeight: 700 }}>Total Amount = {total} â‚¬</Text>
      </Card.Content>
    </>
  );
}
