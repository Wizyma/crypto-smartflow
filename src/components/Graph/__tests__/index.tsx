import { render } from '@testing-library/react';
import React from 'react';

import Graph from 'components/Graph';
import data from '../../../__fixtures__';

test('render Graph component correctly', () => {
  const { asFragment } = render(<Graph {...data} />);

  expect(asFragment()).toMatchSnapshot();
});
