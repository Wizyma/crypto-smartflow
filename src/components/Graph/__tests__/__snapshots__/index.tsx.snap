// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render Graph component correctly 1`] = `
<DocumentFragment>
  <div
    class="jsx-2956695818 content "
  >
    <div
      class="recharts-wrapper"
      style="position: relative; cursor: default; width: 800px; height: 800px;"
    >
      <svg
        class="recharts-surface"
        height="800"
        version="1.1"
        viewBox="0 0 800 800"
        width="800"
      >
        <defs>
          <clippath
            id="recharts1-clip"
          >
            <rect
              height="790"
              width="790"
              x="5"
              y="5"
            />
          </clippath>
        </defs>
        <g
          class="recharts-layer recharts-pie"
        >
          <g
            class="recharts-layer"
          >
            <g
              class="recharts-layer recharts-pie-sector"
            />
            <g
              class="recharts-layer recharts-pie-sector"
            />
            <g
              class="recharts-layer recharts-pie-sector"
            />
            <g
              class="recharts-layer recharts-pie-sector"
            />
            <g
              class="recharts-layer recharts-pie-sector"
            />
          </g>
        </g>
      </svg>
      <div
        class="recharts-tooltip-wrapper"
        style="transform: translate(undefinedpx, undefinedpx); pointer-events: none; visibility: hidden; position: absolute; top: 0px;"
      >
        <div
          class="recharts-default-tooltip"
          style="margin: 0px; padding: 10px; background-color: rgb(255, 255, 255); border: 1px solid #ccc; white-space: nowrap;"
        >
          <p
            class="recharts-tooltip-label"
            style="margin: 0px;"
          />
        </div>
      </div>
    </div>
  </div>
  <div
    class="jsx-3382564211 divider "
    role="separator"
  />
  <div
    class="jsx-2956695818 content "
  >
    <p
      class="jsx-3390086683  "
      style="color: rgb(204, 204, 204); text-align: center; font-weight: 700;"
    >
      Total Amount = 2942.09 â‚¬
    </p>
  </div>
</DocumentFragment>
`;
